src/components/Tabs/index.tsx

```diff
@@ -26,121 +23,76 @@ const Tabs = (props: Props) => {
     tasksAndSors,
     budgetCode,
     workOrder,
+    setActiveTab,
   } = props
 
   const router = useRouter()
 
-  const formatTabNameToId = (tabName) => {
-    if (tabName.includes('tab')) {
-      return tabName
-    }
+  const QUERY_KEY = 'currentTab'
 
-    return `${tabName.replace(/ +/g, '-').toLowerCase()}-tab`
-  }
-  const [activeTab, setActiveTab] = useState(
-    router.asPath.split('#')[1] || formatTabNameToId(tabsList[0])
-  )
+  const [currentTab, setCurrentTab] = useState<TabName>(() => {
+    const firstTab = tabsList[0]
+    const currentQuery = router?.query?.[QUERY_KEY] ?? null
+
+    if (currentQuery === null) return firstTab
+
+    const matchingTabs = Object.keys(TAB_REGISTRY)
+      .map((x) => TAB_REGISTRY[x])
+      .filter((x) => x.id === currentQuery)
+
+    if (matchingTabs.length === 0) return firstTab
 
-  useEffect(() => {
-    if (typeof window !== 'undefined') {
-      require('lbh-frontend').initAll()
-    }
-  }, [])
-
-  const renderTabComponentView = (activeTabId, tabName) => {
-    switch (activeTabId) {
-      case 'work-orders-history-tab':
-        return (
-          <WorkOrdersHistoryView
-            propertyReference={propertyReference}
-            tabName={tabName}
-          />
-        )
-      case 'related-work-orders-tab':
-        return (
-          <RelatedWorkOrdersView
-            workOrderReference={workOrderReference}
-            tabName={tabName}
-          />
-        )
-
-      case 'tasks-and-sors-tab':
-        return (
-          <TasksAndSorsView
-            tabName={tabName}
-            tasksAndSors={tasksAndSors}
-            budgetCode={budgetCode}
-          />
-        )
-      case 'notes-tab':
-        return (
-          <NotesView
-            workOrderReference={workOrderReference}
-            workOrder={workOrder}
-            tabName={tabName}
-            setActiveTab={setActiveTab}
-          />
-        )
-      case 'pending-variation-tab':
-        return <VariationSummaryTab workOrderReference={workOrderReference} />
-      case 'photos-tab':
-        return (
-          <PhotosTab
-            workOrderReference={workOrderReference}
-            tabName={tabName}
-          />
-        )
-      default:
-        return (
-          <WorkOrdersHistoryView
-            propertyReference={propertyReference}
-            tabName={tabName}
-          />
-        )
-    }
+    return matchingTabs[0].name
+  })
+
+  const handleSelectTab = (newTab: TabName) => {
+    if (newTab === currentTab) return
+
+    setCurrentTab(() => newTab)
+
+    router.replace({
+      pathname: router.pathname,
+      query: {
+        ...router.query,
+        [QUERY_KEY]: TAB_REGISTRY[newTab].id,
+      },
+    })
```

Comment:
Replace the url with `?currentTab`. This prevents the browser history from being polluted. The back button will navigate to the previous page. Plus the currently selected tab will be remembered